### DISCUSSION TITLE
Database Issue Running Wiki JS on Namecheap Shared Hosting

### DISCUSSION THREAD

**Discussion by LindForce** [requarks/wiki#3408] [Help / Questions]
I'm trying to set up Wiki JS on my web host through CPanel. I've managed to set up the actual Node application correctly I think, but I'm running into issues when trying to run the index.js file. 

The first time I started the application, I got a "syntax error or access violation" regarding that the "specified key was too long". However, when trying to run the application the second time, as well as after that, I get the following database error:

`migration file "2.0.0.js" failed
migration failed with error: create table `analytics` (`key` varchar(255) not null, `isEnabled` boolean not null                                                                                                                              default '0', `config` json not null) default character set utf8mb4 - Table 'analytics' already exists
{ method: 'update',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings: [ 0 ],
  __knexQueryUid: 'a5e70fd0-c5fe-11ea-aa31-d1791ca8c0b8',
  sql: 'update `migrations_lock` set `is_locked` = ?' }
2020-07-14T18:20:08.404Z [MASTER] error: Database Initialization Error: create table `analytics` (`key` varchar(                                                                                                                             255) not null, `isEnabled` boolean not null default '0', `config` json not null) default character set utf8mb4 -                                                                                                                              Table 'analytics' already exists
{ Error: create table `analytics` (`key` varchar(255) not null, `isEnabled` boolean not null default '0', `confi                                                                                                                             g` json not null) default character set utf8mb4 - Table 'analytics' already exists
    at Function.get (/home/lindfors/wiki.shadowrain.net/node_modules/knex/lib/util/make-knex.js:42:26)
    at Object.exports.up.knex [as up] (/home/lindfors/wiki.shadowrain.net/server/db/migrations/2.0.0.js:9:15)
    at current.then (/home/lindfors/wiki.shadowrain.net/node_modules/knex/lib/migrate/Migrator.js:508:40)
    at process._tickCallback (internal/process/next_tick.js:68:7)
From previous event:
    at runCallback (timers.js:705:18)
    at tryOnImmediate (timers.js:676:5)
    at processImmediate (timers.js:658:5)
From previous event:
    at Object.init (/home/lindfors/wiki.shadowrain.net/server/core/db.js:201:28)
    at Object.init (/home/lindfors/wiki.shadowrain.net/server/core/kernel.js:13:35)
    at Object.<anonymous> (/home/lindfors/wiki.shadowrain.net/server/index.js:33:13)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
  code: 'ER_TABLE_EXISTS_ERROR',
  errno: 1050,
  sqlState: '42S01',
  sqlMessage: 'Table \'analytics\' already exists',
  originalStack:
   'Error: create table `analytics` (`key` varchar(255) not null, `isEnabled` boolean not null default \'0\', `c                                                                                                                             onfig` json not null) default character set utf8mb4 - Table \'analytics\' already exists\n    at Packet.asError                                                                                                                              (/home/lindfors/wiki.shadowrain.net/node_modules/mysql2/lib/packets/packet.js:712:17)\n    at Query.execute (/ho                                                                                                                             me/lindfors/wiki.shadowrain.net/node_modules/mysql2/lib/commands/command.js:28:26)\n    at Connection.handlePack                                                                                                                             et (/home/lindfors/wiki.shadowrain.net/node_modules/mysql2/lib/connection.js:417:32)\n    at PacketParser.Connec                                                                                                                             tion.packetParser.p [as onPacket] (/home/lindfors/wiki.shadowrain.net/node_modules/mysql2/lib/connection.js:75:1                                                                                                                             2)\n    at PacketParser.executeStart (/home/lindfors/wiki.shadowrain.net/node_modules/mysql2/lib/packet_parser.j                                                                                                                             s:75:16)\n    at Socket.Connection.stream.on.data (/home/lindfors/wiki.shadowrain.net/node_modules/mysql2/lib/co                                                                                                                             nnection.js:82:25)\n    at Socket.emit (events.js:198:13)\n    at addChunk (_stream_readable.js:288:12)\n    at                                                                                                                              readableAddChunk (_stream_readable.js:269:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at                                                                                                                              TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)' }`

Full logs here: https://pastebin.com/dc0238mm

**Host Info:**
OS: Cloud Linux OS based on Centos
Wiki.js version: 2.4.107
Database engine: MariaDB 10.3.23

Any help is greatly appreciated! Thanks in advance.

**Comment by Smankusors** [comment]
maybe related to #1113 for "specified key was too long". Can you try solutions from that issue?

**Comment by LindForce** [comment]
> maybe related to #1113 for "specified key was too long". Can you try solutions from that issue?

Thanks, I've checked out the solutions in that thread. Changing the DB collation didn't seem to work, so that leaves me with changing the storage engine from MyISAM to InnoDB. However, problem is that my host doesn't allow me to change the storage engine for the database, other than manually selecting it when creating a table, which isn't really helpful when the error occurs when the tables are created by the app.

**Comment by sapphire008** [comment]
Have you found a solution to this? Thank you.

**Comment by LindForce** [comment]
> Have you found a solution to this? Thank you.

Changing my storage engine from MyISAM to InnoDB was what resolved the issue for me. 

However, I still wasn't able to change that setting with my shared hosting plan, so now I'm hosting it on a machine of my own instead.
